File: 프로그래머스/unrated/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    answer = 0
    lp = len(p)
    num = int(p)
    # t의 부분 문자열 중 p와 길이가 같은 문자열 확인
    for i in range(len(t)-lp+1):
        # t의 부분 문자열이 p보다 작거나 같으면 answer += 1
        if int(t[i:i+lp]) <= num:
            answer += 1
            
    return answer

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        
        long pNum = Long.valueOf(p);      // p가 나타내는 수
        int pLen = p.length();
        
        for (int i = 0; i < t.length() - pLen + 1; i++) {
            String tPartial = t.substring(i, i + pLen);
            long tNum = Long.valueOf(tPartial);     // p의 최대 길이는 10,000
            
            if (tNum <= pNum) {
                answer++;
            }
        }
        
        
        return answer;
    }
}